### DISCUSSION TITLE
Set passcode

### DISCUSSION THREAD

**Discussion by Umar-37** [jbergler/hass-ttlock#120] [Q&A] [answered]
First, thank you for your great work on this plugin.

This is my first time using Home Assistant, so I apologize if I messed anything up. I have a TTLock lock that I can successfully unlock and lock through your plugin. However, my question is: how do I set a passcode through the plugin? I even tried through the automation, but the only options I have are lock and unlock.

If you have any guide, that would be a great help.

**Comment by jbergler** [comment, marked as answer]
Hiya - the functionality is exposed as a home assistant service, the docs for which are here: https://www.home-assistant.io/docs/scripts/service-calls/

In the UI you can find these via Developer Tools > Services > ttlock.* and the UI will give you templates for the required params.
The spec is located here: https://github.com/jbergler/hass-ttlock/blob/develop/custom_components/ttlock/services.yaml
